connector/connect/common/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -236,6 +237,9 @@ message StreamingQueryInstanceId {
   // will generate a unique run_id. Therefore, every time a query is restarted from
   // checkpoint, it will have the same `id` but different `run_id`s.
   string run_id = 2;
+
+  // (Optional) The name of this query.
+  optional string name = 3;
```

Comment:
Why is this needed. Note that query-name is not part of real identity of the query.  This can be extra field like response for 'WriteStreamOperation'.

Reply:
Right I'm also not very sure here. Basically I added this because here:
https://github.com/apache/spark/blob/069caf4dcdb4380cd6eeca394b55824d5b71940a/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala#L2411
In the `sqm.get(queryID)` handler, the query may or may not have a name. But in case it has, if we don't return the name then the client won't have it's name.

We could also maintain a local cache of query names, but that would add more complexity on cleaning up that cache though... What do you think

Reply:
We can return the name, but we don't need to change `StreamingQueryInstanceId`. We can return both `StreamingQueryInstanceId` and the name